tests/integration/any_charm.py

```diff
@@ -41,6 +44,49 @@ def update_ingress(self, ingress_config):
         """
         self.ingress.update_config(ingress_config)
 
+    def _has_required_fields(self, rel: Dict) -> bool:
+        """Check for required fields in relation
+
+        Args:
+            rel: relation to check
+
+        Returns:
+            Returns true if all fields exist
+        """
+        return "service-hostname" in rel and "service-name" in rel and "service-port" in rel
+
+    def _has_app_data(self) -> bool:
+        """Check for app in relation data
+
+        Returns:
+            Returns true if app data exist
+        """
+        return self.app in self.model.relations["ingress"][0].data
+
+    def _has_ingress_relation(self) -> bool:
+        """Check for ingress relation
+
+        Returns:
+            Returns true if ingress relation exist
+        """
+        return "ingress" in self.model.relations and len(self.model.relations["ingress"]) > 0
+
+    def ingress_config(self) -> Dict:
+        """Get ingress config from relation or default
+
+        Returns:
+            The ingress config to be used
+        """
+        if self._has_ingress_relation() and self._has_app_data():
+            rel = self.model.relations["ingress"][0].data[self.app]
+            if self._has_required_fields(rel):
+                return {
+                    "service-hostname": rel["service-hostname"],
```

Comment:
Would it be worth making these constants? They seem to be used in a few places

Reply:
Nice :)